frontend/src/Frontend/UI/DeploymentSettings.hs

```diff
@@ -727,12 +727,31 @@ uiMetaData m mTTL mGasLimit = do
       readPact wrapper =  fmap wrapper . readMay . T.unpack
 
 -- | Set the sender to a fixed value
-uiSenderFixed :: DomBuilder t m => AccountName -> m (Dynamic t (Maybe (Some AccountRef)))
-uiSenderFixed sender = do
+uiSenderFixed
+  :: ( DomBuilder t m
+     , HasWallet model key t
+     , HasNetwork model t
+     )
+  => model
+  -> Dynamic t (Maybe ChainId)
+  -> AccountName
+  -> m (Dynamic t (Maybe (Some AccountRef)))
+uiSenderFixed model chainId sender = do
+  -- Look up the account by name for the current network & chain
+  let mAccRef = runMaybeT $ do
+        n <- lift $ model ^. network_selectedNetwork
+        c <- MaybeT chainId
+        MaybeT $ findAccountByName (model ^. wallet) n c sender
+
   _ <- uiInputElement $ def
     & initialAttributes %~ Map.insert "disabled" ""
     & inputElementConfig_initialValue .~ unAccountName sender
-  pure $ pure $ pure $ Some $ AccountRef_Vanity sender "chain" -- TODO check the chain
+    & inputElementConfig_setValue .~ ffor (updated mAccRef) (\case
+        Nothing -> "Account name not found!"
+        Just (Some a) -> accountRefToName a
```

Comment:
Is `accountRefToName a` ever different than `sender` or are just checking existance?

Reply:
By that stage there should be no difference.